### DISCUSSION TITLE
Code generation for sirius-web

### DISCUSSION THREAD

**Discussion by hakut** [eclipse-sirius/sirius-components#3432] [Q&A]
Hello,

I want to build a project that can generate code from my models. I have gotten the feedback that there is no built-in code generation mechanism exists in sirius-web. Can anyone help me to integrate this functionality to sirius-web? I need a little guidance.

Thank you in advance.

**Comment by baristezel** [comment]
As you said, there is no built-in code generation or transformation mechanism. So best way i think that you can use abilities of sirius desktop such as Acceleo. You have to create immigrated version of your tool in sirius desktop. You can do it both way. First one is creating siruis desktop based tool and transfer it to siruis web by tracing steps at the document. (I haven't achived it yet :D ) Other way is creating twin meta model at the desktop and passing your Siruis web instances to the Siruis desktop to use code generation mechanism in their. Instances are so similiar so with little adjustment, you can use your instances in both environment. To more details, do not hesitate to contect me personally.

**Reply by hakut** [reply]
Thank you for your help. I'll try to achieve code generation in sirius-web. I'll try my best.

**Comment by pcdavid** [comment]
Sirius Web stores its semantic data as EMF models, so in the backend side you can use any EMF-compatible technology to generate code (or text in general) from the models, including for example [Acceleo](https://eclipse.dev/acceleo/).

It's possible to do code generation, but there are many options, the best one will depend a lot on your concrete use case. In particular, the approach will probably be very different between "batch" generation (possibly long-running generators that will create a large volume of code) or "live" (and fast) generation of a single file.

**Reply by hakut** [reply]
Thank you for your help. Live generation is enough for me at the moment. So I'll work on that.

**Comment by ebousse** [comment]
Hi there, sorry to dig this thread 2 years later :). 

Is there new advice on this matter? I was wondering how to proceed if we would like to deploy a code generator (or a model interpreter) within a Studio made using Sirius web.
